Assets/Scripts/Models/Furniture.cs

```diff
@@ -541,62 +561,124 @@ public void ReadXmlParams(XmlReader reader)
         {
             do
             {
-                string k = reader.GetAttribute("name");
-                float v = float.Parse(reader.GetAttribute("value"));
-                furnParameters[k] = v;
+                string name = reader.GetAttribute("name");
+                string value = reader.GetAttribute("value");
+                // "type" attirubte is optional, defaults to float
+                string type = reader.GetAttribute("type");
+                if(type == null || type == "")
+                {
+                    type = "float";
+                }
+                AddParameter(name, value, type);
             } while (reader.ReadToNextSibling("Param"));
         }
     }
 
+    void AddParameter(string name, string value, string type)
+    {
+        switch (type)
+        {
+            case "string":
+            case "float":
+                SetParameter(name, value);
+                break;
+            default:
+                Debug.LogError(string.Format("Invalid type \"{0}\" for parameters.", type));
+                break;
+        }
+    }
+
     /// <summary>
     /// Gets the custom furniture parameter from a string key.
     /// </summary>
     /// <returns>The parameter value (float).</returns>
     /// <param name="key">Key string.</param>
     /// <param name="default_value">Default value.</param>
-    public float GetParameter(string key, float default_value)
+    public string GetParameter(string key, string default_value = null)
     {
         if (furnParameters.ContainsKey(key) == false)
         {
+            if (default_value == null) return "";
             return default_value;
         }
 
-        return furnParameters[key];
+        return furnParameters[key].value;
+        //return (T) T.parse(furnParameters[key].value);
     }
 
-    public float GetParameter(string key)
+    /// <summary>
+    /// Gets the custom furniture parameter from a string key.
+    /// </summary>
+    /// <returns>The parameter value (float).</returns>
+    /// <param name="key">Key string.</param>
+    /// <param name="default_value">Default value.</param>
+    public float GetParameterAsFloat(string key, float default_value = 0f)
     {
-        return GetParameter(key, 0);
-    }
+        if (furnParameters.ContainsKey(key) == false)
+        {
+            return default_value;
+        }
 
+        return float.Parse(furnParameters[key].value);
```

Comment:
Maybe check if the parameter is of type float first? I also don't really like the fact that there will be more functions per type of parameter (not sure if there is an alternative).